packit_service/worker/handlers/testing_farm_handlers.py

```diff
@@ -102,6 +105,35 @@ def run(self) -> TaskResults:
             copr_build = CoprBuildModel.get_by_id(self.build_id)
         else:
             copr_build = testing_farm_helper.get_latest_copr_build(target=self.chroot)
+
+        # If no suitable copr build is found, run trigger copr build
+        if (
+            not copr_build
+            or copr_build.commit_sha != self.data.commit_sha
+            or copr_build.status != PG_COPR_BUILD_STATUS_SUCCESS
+        ):
+
+            logger.info("No suitable copr-build found, run copr build.")
+
+            result_details = {
+                "msg": "Build required, triggering the build",
+                "event": self.data,
+                "package_config": self.package_config,
+                "job": self.job_config.type.value if self.job_config else None,
+                "job_config": dump_job_config(self.job_config),
+            }
+
+            CoprBuildJobHelper(
+                self.service_config,
+                self.package_config,
+                self.project,
+                self.data,
+                self.db_trigger,
+                self.job_config,
+            ).run_copr_build()
+
+            return TaskResults(success=True, details=result_details)
```

Comment:
This is tricky:
* `CoprBuildJobHelper.run_copr_build` returns `TaskResult` so we might want to reuse it (and maybe enhance to mention why this has been triggered).
* `CoprBuildJobHelper.run_copr_build` can fail, or in other words, it can return `TaskResult` with `success=False`.

Reply:
Based on our [previous discussion](https://github.com/packit/packit-service/issues/1036#issuecomment-853743715), I assumed that triggering a new copr build and then return `TaskResult` with `success=True` is ok.

But maybe it'd be better for TF to return `TaskResult` with `success=False` always when triggering new copr build.
After all, TF was not successfull.

I'm still not sure how does this work as a whole. Is TF triggered when copr build  finishes successfully? If yes, then after the new copr build is ready second instance of TF is triggered and it should find appropriate copr build, run tests and return proper result. If the copr build fails, second instance of TF is not triggered - having successful result from the first instance of TF  probably isn't what we want.

Does the above make at least a bit of sense, or I'm completely lost?

Reply:
> Does the above make at least a bit of sense, or I'm completely lost?

No, not at all..;)

> s TF triggered when copr build finishes successfully? If yes, then after the new copr build is ready second instance of TF is triggered and it should find appropriate copr build, run tests and return proper result.

Yes, exactly.

> If the copr build fails, second instance of TF is not triggered - having successful result from the first instance of TF probably isn't what we want.

Here, we are speaking about two things -- task result (=status for service) and build result. (=status for user).
I previously thought that we will not submit the actual build, but add it to the queue -- so we would be able to return successful task result (and do nothing else).

Here, I would return the same result we get from `CoprBuildJobHelper.run_copr_build` and attach some note about the reason.